### PR TITLE
mchrusci/ci-watchdog

### PR DESCRIPTION
**Pull request by mchrusci**
PR for testing, please ignore.

### CODE REVIEW

**Comment by mchrusci** [comment]
TEST NOW

File: .ci/watchdog/ci_watchdog.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        current_prs.append(str(pr.number))
+        log.info("Checking PR#%s", pr.number)
+        last_commit = (pr.get_commits().reversed)[0]
+        statuses = last_commit.get_statuses()
+        # Filter statuses to contain only those related to Jenkins CI and check if CI in Jenkins started
+        jenk_statuses = [stat for stat in statuses if "Jenkins CI" in stat.context]
+        other_statuses = [stat for stat in statuses if not "Jenkins CI" in stat.context]
+        if not jenk_statuses and (now_time - pr.updated_at > awaiting_jenkins_treshold):
+            config = communicate_fail("Jenkins CI report for PR#{} not present on GitHub!".format(pr.number), pr, slack_app, config)
+        for stat in jenk_statuses:
```
Comment (ldurka) [inline .ci/watchdog/ci_watchdog.py]: a little bit long for body

**Comment by mchrusci** [comment]
To be refactored

File: .ci/watchdog/ci_watchdog.groovy [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                dir ("$BUILD_WORKSPACE") {
+                    checkout([$class: 'GitSCM', branches: [[name: "*/$BRANCH"]], 
+                            doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', timeout: 30]],  submoduleCfg: [],
+                            userRemoteConfigs: [[credentialsId: '6887a177-8c4d-4fe9-9c3b-fcd71b22bfba', url: 'git@github.com:NervanaSystems/ngraph-onnx.git']]])
+                }
+            }
+            stage("Prepare environment") {
+                dir ("$BUILD_WORKSPACE") {
+                    sh """#!/bin/bash
+                        virtualenv -p python2 venv
```
Comment (postrational) [inline .ci/watchdog/ci_watchdog.groovy]: Let's avoid using Python2. End of life for Python 2 is in a little over a year, we're hoping this code will live longer.
https://pythonclock.org/

File: .ci/watchdog/ci_watchdog.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/python2.7
+
+# INTEL CONFIDENTIAL
+# Copyright 2017 Intel Corporation All Rights Reserved.
```
Comment (postrational) [inline .ci/watchdog/ci_watchdog.py]: 2018

**Pull request closed without merging**